pyomo/repn/plugins/nl_writer.py

```diff
@@ -1576,6 +1877,16 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
+    def __eq__(self, other):
```

Comment:
Are we okay making this class not hashable (I'm fine with it; I just wanted to make sure)?

Reply:
I needed it comparable for testing, but that is a good point on hashability.  I added a reasonable implementation of `__hash__`.